### ISSUE TITLE
Importing @tanstack/react-query forces react-native as a peer dependency, and doesn't work

### ISSUE THREAD

**Issue by bchilcott** [stackblitz/core#2522]
### **Description of Bug**
When trying to import `@tanstack/react-query`, the `react-native` library gets installed as a peer dependency. Then, when trying to use the library in the code, some imports don't exist. I assume an incorrect version is being used, although it claims to be up to date in the dependency list (4.29.7 at the time of writing).

https://stackblitz.com/fork/react-ts

### **Steps to Reproduce**

1. Start a new react project: https://stackblitz.com/fork/react-ts
2. Add `@tanstack/react-query` as a dependency
3. react-native is forced as a peer dependency
4. Try to `import {QueryClient} from "@tanstack/react-query"` in `App.tsx`. The library supposedly has no exported member with that name.

### **Expected Behavior**
The `@tanstack/react-query` library should work as-per the [Tanstack Query Docs](https://tanstack.com/query/v4/docs/react/quick-start).

### **Screenshots/Screencast**
![image](https://github.com/stackblitz/core/assets/38051283/f15d79c4-f72d-424a-b73f-31fd5fd84057)

![image](https://github.com/stackblitz/core/assets/38051283/7bde0f77-17c0-4ba2-8b7a-f971622cacf5)

**Comment by fvsch** [comment]
Looks like this package has `react-native` in its peerDependencies:

```json
{
  "peerDependencies": {
    "react": "^16.8.0 || ^17.0.0 || ^18.0.0",
    "react-dom": "^16.8.0 || ^17.0.0 || ^18.0.0",
    "react-native": "*"
  },
  "peerDependenciesMeta": {
    "react-dom": {
      "optional": true
    },
    "react-native": {
      "optional": true
    }
  }
}
```

And our package resolution logic might not support [peerDependenciesMeta](https://docs.npmjs.com/cli/v7/configuring-npm/package-json#peerdependenciesmeta).

**Comment by fvsch** [comment]
The part about the `QueryClient` export seems to be a separate issue. It could be that our TypeScript setup doesn't like that https://unpkg.com/browse/@tanstack/react-query@4.29.7/build/lib/index.d.ts re-exports types from a different package.

As a workaround you can probably use

```
import type { QueryClient } from "@tanstack/query-core'"
```

or if you're importing the actual class:

```
import { QueryClient } from "@tanstack/query-core'"
```

**Comment by bchilcott** [comment]
> Looks like this package has react-native in its peerDependencies:

Good point, I hadn't spotted that. This probably isn't as problematic as I thought it was then, just a very minor inconvenience.

> import { QueryClient } from "@tanstack/query-core'"

This doesn't seem to work, the module can't be found even after adding to the project directly. Have you got an example of this working?
